core/tee/tee_svc_storage.c

```diff
@@ -598,7 +663,15 @@ TEE_Result tee_svc_storage_obj_create(uint32_t storage_id, void *object_id,
 		goto exit;
 	}
 
+clean:
+	tee_file_ops.unlink(initfile);
+
 exit:
+	free(file);
+	file = NULL;
+	free(initfile);
+	initfile = NULL;
+
 	if (res != TEE_SUCCESS) {
 		if (fd >= 0)
 			tee_file_ops.close(fd);
```

Comment:
Update: After a chat with Cédric, I was wrong in my remark about `po`.

However, I still feel the code hard to read.